flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamScanSql.java

```diff
@@ -248,4 +267,54 @@ public void testConsumeFromStartSnapshotId() throws Exception {
     }
     result.getJobClient().ifPresent(JobClient::cancel);
   }
+
+  @Test
+  public void testConsumeFromStartTag() throws Exception {
+    sql("CREATE TABLE %s (id INT, data VARCHAR, dt VARCHAR)", TABLE);
+    Table table = validationCatalog.loadTable(TableIdentifier.of(icebergNamespace, TABLE));
+
+    // Produce two snapshots.
+    Row row1 = Row.of(1, "aaa", "2021-01-01");
+    Row row2 = Row.of(2, "bbb", "2021-01-01");
+    insertRows(table, row1);
+    insertRows(table, row2);
+
+    String tagName = "t1";
+    long startSnapshotId = table.currentSnapshot().snapshotId();
+    table.manageSnapshots().createTag(tagName, startSnapshotId).commit();
+
+    Row row3 = Row.of(3, "ccc", "2021-01-01");
+    Row row4 = Row.of(4, "ddd", "2021-01-01");
+    insertRows(table, row3, row4);
+
+    TableResult result =
+        exec(
+            "SELECT * FROM %s /*+ OPTIONS('streaming'='true', 'monitor-interval'='1s', "
+                + "'start-tag'='%s')*/",
+            TABLE, tagName);
+    try (CloseableIterator<Row> iterator = result.collect()) {
+      // The row2 in start snapshot will be excluded.
```

Comment:
nit: we can simply say the start snapshot is exclusive. mentioning row2  can be less clear.